### ISSUE TITLE
Entity descriptor signing (1. part of Registry Register Entity txn flow) doesn't work with version 2.1.0

### ISSUE THREAD

**Issue by tjanez** [Zondax/ledger-oasis#126]
This appears to be a regression from the current latest version published on Ledger Live (version 1.8.2).

## Steps to reproduce
```bash
cd $(mktemp -d)

OASIS_CORE_VERSION=21.2.7
wget https://github.com/oasisprotocol/oasis-core/releases/download/v${OASIS_CORE_VERSION}/oasis_core_${OASIS_CORE_VERSION}_linux_amd64.tar.gz
tar -xf oasis_core_${OASIS_CORE_VERSION}_linux_amd64.tar.gz
export OASIS_NODE="$PWD/oasis_core_${OASIS_CORE_VERSION}_linux_amd64/oasis-node"

OASIS_CORE_LEDGER_VERSION=1.2.0
wget https://github.com/oasisprotocol/oasis-core-ledger/releases/download/v${OASIS_CORE_LEDGER_VERSION}/oasis_core_ledger_${OASIS_CORE_LEDGER_VERSION}_linux_amd64.tar.gz
tar -xf oasis_core_ledger_${OASIS_CORE_LEDGER_VERSION}_linux_amd64.tar.gz
export LEDGER_SIGNER_PATH="$PWD/oasis_core_ledger_${OASIS_CORE_LEDGER_VERSION}_linux_amd64/ledger-signer"

wget https://github.com/oasisprotocol/mainnet-artifacts/releases/download/2021-04-28/genesis.json -O genesis.json

ENTITY_DIR=entity0/
LEDGER_SIGNER_FLAGS=(
  --signer.dir $ENTITY_DIR
  --signer.backend plugin
  --signer.plugin.name ledger
  --signer.plugin.path "$LEDGER_SIGNER_PATH"
)
TX_FLAGS=(
  --genesis.file genesis.json
  "${LEDGER_SIGNER_FLAGS[@]}"
)

# Open Oasis app on Ledger.

mkdir $ENTITY_DIR
$OASIS_NODE signer export "${LEDGER_SIGNER_FLAGS[@]}"

$OASIS_NODE registry entity gen_register \
  "${TX_FLAGS[@]}" \
  --transaction.file tx_entity_registery.json \
  --transaction.nonce 42 \
  --transaction.fee.gas 1000 \
  --transaction.fee.amount 2000
```

## Results with Version 1.8.2
```
$ $OASIS_NODE registry entity gen_register \
  "${TX_FLAGS[@]}" \
  --transaction.file tx_entity_registery.json \
  --transaction.nonce 42 \
  --transaction.fee.gas 1000 \
  --transaction.fee.amount 200
You are about to sign the following transaction:
  Method: registry.RegisterEntity
  Body:
    {
      "untrusted_raw_value": {
        "v": 2,
        "id": "l+cuboPsOeuY1+kYlROrpmKgiiELmXSw9xl0WEg8cWE="
      },
      "signature": {
        "public_key": "l+cuboPsOeuY1+kYlROrpmKgiiELmXSw9xl0WEg8cWE=",
        "signature": "ijDIwV72NpgiaSyR4rMfexdmtCfRh4VTBqkMeE8D7vSA/AtLMr9mgcr3k9PRlU96zreva0W7BqKsN1jM0aoKCA=="
      }
    }
  Nonce:  42
  Fee:
    Amount: 0.000002 ROSE
    Gas limit: 1000
    (gas price: 0.000000002 ROSE per gas unit)
Other info:
  Genesis document's hash: 53852332637bacb61b91b6411ab4095168ba02a50be4c3f82448438826f23898

You may need to review the transaction on your device if you use a hardware-based signer plugin...
```

## Results with Version 2.1.0

NOTE: I've compiled the latest commit on the `dev` branch as of today: f8a4615615fcb078123a5bf4efb0eec5eb833398.

```
$ $OASIS_NODE registry entity gen_register \
>   "${TX_FLAGS[@]}" \
>   --transaction.file tx_entity_registery.json \
>   --transaction.nonce 42 \
>   --transaction.fee.gas 1000 \
>   --transaction.fee.amount 2000
ts=2021-07-09T16:55:43.717754763Z level=error module=cmd/registry/entity caller=entity.go:297 msg="failed to sign entity descriptor" err="signature/signer/plugin: failed to sign: ledger: failed to sign message: ledger/oasis: failed to sign: Unrecognized error code"
```

**Comment by emmanuelm41** [comment]
I added a zemu test case. I used the the value I found on the file called `tx_entity_registery.json`. This file is created after signing the tx with app 1.8.2.

The content is:
```
{
"untrusted_raw_value":"pGNmZWWiY2dhcxkD6GZhbW91bnRCB9BkYm9keaJpc2lnbmF0dXJlomlzaWduYXR1cmVYQIowyMFe9jaYImkskeKzH3sXZrQn0YeFUwapDHhPA+70gPwLSzK/ZoHK95PT0ZVPes63r2tFuwairDdYzNGqCghqcHVibGljX2tleVggl+cuboPsOeuY1+kYlROrpmKgiiELmXSw9xl0WEg8cWFzdW50cnVzdGVkX3Jhd192YWx1ZVgpomF2AmJpZFggl+cuboPsOeuY1+kYlROrpmKgiiELmXSw9xl0WEg8cWFlbm9uY2UYKmZtZXRob2R3cmVnaXN0cnkuUmVnaXN0ZXJFbnRpdHk=",
"signature":{
        "public_key":"l+cuboPsOeuY1+kYlROrpmKgiiELmXSw9xl0WEg8cWE=",
        "signature":"vg6hlN+4beAmOdarYQhbl4rcijcErUBl0WkUKa77rzdqGi1kfWEw70EDYtIHwrsNL30Ihk6wBa/fmjJla6nBDw=="
      }
 }
```

The zemu test passed, so I guess the issue is one step before signing the tx. Is it possible @tjanez @jleni ?

**Comment by tjanez** [comment]
> The zemu test passed, so I guess the issue is one step before signing the tx. Is it possible @tjanez @jleni ?

Indeed, that is the case here. The `registry.RegisterEntity` txn signing is special because it involves 2 signatures:

1. First, the user needs to sign the entity's descriptor (CLI shows no preview for that currently)
2. Then, the user needs to sign the `registry.RegisterEntity` txn (CLI shows preview as usual) which contains the signed entity descriptor as part of the payload.

So, the regression happens when trying to sign the entity's descriptor (i.e. the 1st signature).

**Comment by tjanez** [comment]
Fixed in #129.
